src/contexts/AssetContext/Provider.tsx

```diff
@@ -1,18 +1,18 @@
 import React, { useCallback, useEffect, useState } from "react";
-import { IAsset, IGetAccountAssetsResult, IGetAssetResult } from "types/NXTAPI";
+import { IAccountAsset, IGetAccountAssetsResult, IGetAssetResult } from "types/NXTAPI";
 import useAccount from "hooks/useAccount";
 import useAPI from "hooks/useAPI";
 import useBlocks from "hooks/useBlocks";
 import Context from "./Context";
 
 const AssetProvider: React.FC = ({ children }) => {
-  const [heldAssets, setHeldAssets] = useState<Array<IAsset>>();
+  const [heldAssets, setHeldAssets] = useState<Array<IAccountAsset>>();
```

Comment:
also noticing that this will run every block. which happens quite often. so it may be wise to add an issue to review the app performance from a render perspective. otherwise the setter here will cause a re-render each block due to it being an array

Reply:
those renders (from the portfolio page) only take about 20 ms and they're required because if we don't update the held assets on each block then we won't display incoming/outgoing assets in the next block. while it's possible to improve performance by only doing it every other block or something, that creates a poor experience where the wallet has a delay vs the blockchain. 

I am aware of another performance issue that I just started noticing though so I did make an issue on that: https://github.com/Up-North-Coding/jovian/issues/119